src/containers/BooksList.js

```diff
@@ -1,37 +1,47 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import PropTypes from 'prop-types';
 import Book from '../components/Book';
+import CategoryFilter from '../components/CategoryFilter';
+import { setCategoryFilter } from '../actions';
 
-const BookList = ({ books }) => {
-  const showAllBooks = books.map(book => <Book key={book.id} book={book} />);
-
+const BookList = props => {
+  const { books } = props;
+  const handleBookByFilterCategotyChange = props.Filterchanged;
   return (
-    <main className="">
-      <h1 className="">Books Management</h1>
-      <table className="">
+    <div>
+      <CategoryFilter
+        handleBookByFilterCategotyChange={handleBookByFilterCategotyChange}
+      />
+      <table>
         <thead>
           <tr>
-            <th>Title</th>
-            <th>Category</th>
+            <th>title</th>
+            <th>category</th>
           </tr>
         </thead>
-        <tbody>{showAllBooks}</tbody>
+        <tbody>
+          {books.map((book, bookID) => (
+            <Book key={bookID} book={book} />
+          ))}
+        </tbody>
       </table>
-    </main>
+    </div>
   );
 };
-
-BookList.propTypes = {
-  books: PropTypes.arrayOf(
-    PropTypes.shape({
-      id: PropTypes.number,
-      title: PropTypes.string,
-      category: PropTypes.string,
-    }).isRequired,
-  ).isRequired,
+const mapStateToProps = state => {
+  const { books, filter } = state;
+  if (filter !== '') {
+    books = books.filter(book => filter === book.category);
+  }
```

Comment:
wonderful work doing this here instead of in the component 👏  but it breaks.

When i try to change category

![Screenshot 2019-12-11 at 10 39 25 AM](https://user-images.githubusercontent.com/10922508/70609983-0e61ee00-1c03-11ea-89cc-879fdee09e82.png)


Try to update the code like so:

```
const mapStateToProps = state => {
  const { books, filter } = state;
 
  return {
   books: filter === '' ? books : books.filter(book => filter === book.category),

  };
};

```